controllers/eventingauth_controller.go

```diff
@@ -0,0 +1,148 @@
+// +kubebuilder:rbac:groups=operator.kyma-project.io,resources=eventingauths/finalizers,verbs=update
+func (r *eventingAuthReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling EventingAuth", "name", req.Name, "namespace", req.Namespace)
+
+	cr, err := fetchEventingAuth(ctx, r.Client, req.NamespacedName)
+	if err != nil {
+		logger.Info("EventingAuth not found", "name", req.Name, "namespace", req.Namespace)
+		return ctrl.Result{}, err
+	}
+
+	// TODO: Use correct pointing to the target cluster
+	appSecretExists, err := hasTargetClusterApplicationSecret(ctx, r.Client)
+	if err != nil {
+		logger.Error(err, "Failed to retrieve secret state from target cluster", "eventingAuth", cr.Name, "eventingAuthNamespace", cr.Namespace)
+		return ctrl.Result{
+			RequeueAfter: requeueAfterError,
+		}, err
+	}
+
+	if !appSecretExists {
+
+		// TODO: Name of the IAS application should be taken from Kyma CR owner reference
+		iasApplication, err := r.IasClient.CreateApplication(ctx, fmt.Sprintf("eventing-auth-manager-%s", uuid.New()))
+		if err != nil {
+			logger.Error(err, "Failed to create IAS application", "eventingAuth", cr.Name, "eventingAuthNamespace", cr.Namespace)
+			return ctrl.Result{
+				RequeueAfter: requeueAfterError,
+			}, err
+		}
+
+		appSecret := iasApplication.ToSecret(applicationSecretName, applicationSecretNamespace)
+
+		// TODO: Create secret on target cluster by reading the kubeconfig from the secret using the name of the Kyma CR owner reference
+		err = r.Client.Create(ctx, &appSecret)
+		if err != nil {
+			logger.Error(err, "Failed to create application secret", "eventingAuth", cr.Name, "eventingAuthNamespace", cr.Namespace)
+			return ctrl.Result{
+				RequeueAfter: requeueAfterError,
+			}, err
+		}
+	}
+
+	if err := updateStatus(ctx, r.Client, cr, operatorv1alpha1.StateOk); err != nil {
+		logger.Error(err, "Failed to update status of EventingAuth", "name", cr.Name, "namespace", cr.Namespace)
+		return ctrl.Result{
+			RequeueAfter: requeueAfterError,
+		}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func fetchEventingAuth(ctx context.Context, c client.Client, name types.NamespacedName) (operatorv1alpha1.EventingAuth, error) {
+	var cr operatorv1alpha1.EventingAuth
+	err := c.Get(ctx, name, &cr)
+	if err != nil {
+		return cr, err
+	}
+	return cr, nil
+}
+
+func hasTargetClusterApplicationSecret(ctx context.Context, c client.Client) (bool, error) {
+	var s v1.Secret
+	err := c.Get(ctx, client.ObjectKey{
+		Name:      applicationSecretName,
+		Namespace: applicationSecretNamespace,
+	}, &s)
+
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *eventingAuthReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
```

Comment:
I think, we should react to the changes of the owned secret in case changed/deleted. I think we should watch secret with event filters to filter out not relevant secrets instances.